### PR TITLE
feat: 🎸 support remove/duplicate route

### PR DESCRIPTION
**Pull request by khanhx**

### CODE REVIEW

File: api/handler.go [hunk 1]
```diff
@@ -58,6 +58,33 @@ func (s *Server) PatchRouteHandler(w http.ResponseWriter, r *http.Request) {
 	response(w, http.StatusOK, nil)
 }
 
+func (s *Server) DeleteRouteHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (tuongaz) [inline api/handler.go:61]: Please add unit tests

File: api/handler.go [hunk 2]
```diff
@@ -58,6 +58,61 @@ func (s *Server) PatchRouteHandler(w http.ResponseWriter, r *http.Request) {
...
+
+	if err := s.db.DeleteRoute(r.Context(), mockID, routeID); err != nil {
+		responseError(w, http.StatusInternalServerError, err)
+		return
+	}
+
+	response(w, http.StatusOK, nil)
+}
+
+func (s *Server) CreateRouteHandle(w http.ResponseWriter, r *http.Request) {
```
Comment (tuong-at-wes) [inline api/handler.go:73]: `CreateRouteHandler` not `Handle`

File: api/handler.go [hunk 3]
```diff
@@ -58,6 +58,61 @@ func (s *Server) PatchRouteHandler(w http.ResponseWriter, r *http.Request) {
...
+
+	if err = s.db.CreateRoute(r.Context(), mockID, *newRoute); err != nil {
+		responseError(w, http.StatusInternalServerError, err)
+		return
+	}
+
+	response(w, http.StatusOK, *newRoute)
+}
+
+func (s *Server) DuplicateRouteHandle(w http.ResponseWriter, r *http.Request) {
```
Comment (tuong-at-wes) [inline api/handler.go:95]: `DuplicateRouteHandler`

File: api/handler.go [hunk 4]
```diff
@@ -58,6 +58,61 @@ func (s *Server) PatchRouteHandler(w http.ResponseWriter, r *http.Request) {
...
+}
+
+func (s *Server) CreateRouteHandle(w http.ResponseWriter, r *http.Request) {
+	mockID := mux.Vars(r)["mock_id"]
+	data, err := io.ReadAll(r.Body)
+	if err != nil {
+		responseError(w, http.StatusInternalServerError, err)
+	}
+
+	newRoute, err := mock.Route{}.PatchString(string(data))
```
Comment (tuong-at-wes) [inline api/handler.go:80]: Use `json.Unmarshal` function

File: api/handler.go (hunk 2 above)
Comment (tuongaz) [inline api/handler.go:73]: Handler, not Handle

File: api/handler.go (hunk 3 above)
Comment (tuongaz) [inline api/handler.go:95]: Handler, not Handle

File: api/handler.go (hunk 4 above)
Comment (tuongaz) [inline api/handler.go:80]: Pls use `json.Unmarshal`

File: engine/mock/mock.go [hunk 5]
```diff
@@ -117,3 +135,38 @@ func addIDs(m *Mock) {
...
+
+func jsonFieldName(field reflect.StructField) string {
+	name := strings.Split(field.Tag.Get("json"), ",")[0]
+	if name == "" {
+		name = field.Name
+	}
+	return name
+}
+
+func patchStruct(resource interface{}, patches map[string]*json.RawMessage) error {
```
Comment (tuongaz) [inline engine/mock/mock.go:147]: we dont need this if we use the `json.Marshal` above

**Review by tuongaz — APPROVED** [review]

**Pull request merged**